### PR TITLE
Capstone JS Project

### PR DESCRIPTION
**Pull request by thankgodr**
This PR contains the required capstone javascript project that includes the following 

- API calls to the mealDB
- API calls to involvement API
- Test case that covers the fetch and display of meal
- Test that covers the comment counters and Meal counter
- Test that covers the post likes and gets likes
- Gitflow
- UI interaction with popup displays
- POST  and GET comments to and from involvement API
- POST and GET likes to and from involvement API

### CODE REVIEW

File: src/js/modules/comments.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import commentsCounter from './comment_counter';
+
+const baseURL = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/';
+const appID = '9JcQwe8YeiV9ciPMfpK9';
```
Comment (aimalamiri) [inline src/js/modules/comments.js]: The app_id is a type of secret information that should not be committed into the public repository, it's highly recommended to put such a thing in a `.env` file and add enough description on the Readme file about where and how to create and use one.

File: src/js/modules/network/requests/likes_request.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import NetWorkRequest from '../network_request';
+
+export default class LikesRequest {
+    #networRequest;
+
+    #path;
+
+    constructor(gammeID = '9JcQwe8YeiV9ciPMfpK9') {
```
Comment (aimalamiri) [inline src/js/modules/network/requests/likes_request.js]: Again, avoiding putting the secret keys in your public repositories is highly recommended. Putting such a variable into a `.env` file could be significant for security.

I'd recommend the [Dotenv](https://www.npmjs.com/package/dotenv) package, but you're free to use whatever you want.

File: README.md [hunk 3]
```diff
@@ -1,232 +1,98 @@
...
 
 ## 🚀 Live Demo <a name="live-demo"></a>
 
-> Add a link to your deployed project.
+> Available on the link beloow.
 
-- [Live Demo Link](https://yourdeployedapplicationlink.com)
-
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
+- [Live Demo Link](https://thangodr.github.io/capstone-js-project)
```
Comment (aimalamiri) [inline README.md]: [Optional] It's highly recommended to deploy your application so that others can simply check what you've built, but the provided demo link is not working.

**Review by aimalamiri — CHANGES_REQUESTED** [review]
Hi @thankgodr and @momotsow, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

AlmostThereReachGIF

## Highlights

- The application is designed very well and looks fantastic ✔️ ⭐
- Good job for using the right GitHub Flow 💯
- Awesome file organization 🗃️
- Commits are descriptive and meaningful 👍
- Linters are properly set up and passing 🟢 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/tests/comment-counter.test.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      comment: 'taste good',
+      username: 'richard',
+      creation_date: '2022-11-24',
+    },
+    {
+      creation_date: '2022-11-24',
+      comment: 'me',
+      username: 'comment',
+    },
+  ];
```
Comment (jerryowusu) [inline src/js/tests/comment-counter.test.js:26]: - [ ] [OPTIONAL] Make sure that you have tests for all edge cases. You should think at least about the scenario in which there are 0 items on the page (counter function should return 0)

File: src/js/modules/comments.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import commentsCounter from './comment_counter';
+
+const baseURL = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/';
+const appID = process.env.APP_ID;
```
Comment (jerryowusu) [inline src/js/modules/comments.js:4]: Great work in putting the appID in the .env file as requested by the previous reviewer. Unfortunately  I am unable to access it anywhere in your pull request or readme file in order to review your code. You can consider making it accessible to the reviewer by adding it to the `additional information` section when requesting for reviewing or in the pull request summary :pray:

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi Team, :wave:  

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close

## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @jerryowusu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
